pkg/reconcilermanager/controllers/reconciler_container_log_level.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
+	"kpt.dev/configsync/pkg/metrics"
+	"kpt.dev/configsync/pkg/reconcilermanager"
+)
+
+// ReconcilerContainerLogLevelDefaults are the default log level to use for the
+// reconciler deployment containers.
+// All containers default value are 0 excep git-sync which default value is 5
+func ReconcilerContainerLogLevelDefaults() map[string]v1beta1.ContainerLogLevelOverride {
+	return map[string]v1beta1.ContainerLogLevelOverride{
+		reconcilermanager.Reconciler: {
+			ContainerName: reconcilermanager.Reconciler,
+			LogLevel:      0,
+		},
+		reconcilermanager.HydrationController: {
+			ContainerName: reconcilermanager.HydrationController,
+			LogLevel:      0,
+		},
+		reconcilermanager.OciSync: {
+			ContainerName: reconcilermanager.OciSync,
+			LogLevel:      0,
+		},
+		reconcilermanager.HelmSync: {
+			ContainerName: reconcilermanager.HelmSync,
+			LogLevel:      0,
+		},
+		reconcilermanager.GitSync: {
+			ContainerName: reconcilermanager.GitSync,
+			LogLevel:      5, // unlike other containers, git-sync default is 5
+		},
+		reconcilermanager.GCENodeAskpassSidecar: {
+			ContainerName: reconcilermanager.GCENodeAskpassSidecar,
+			LogLevel:      0,
+		},
+		metrics.OtelAgentName: {
+			ContainerName: metrics.OtelAgentName,
+			LogLevel:      0,
+		},
+	}
+}
+
+// setContainerLogLevelDefaults will compile the default and override value of container log level
+func setContainerLogLevelDefaults(overrides []v1beta1.ContainerLogLevelOverride, defaultsMap map[string]v1beta1.ContainerLogLevelOverride) []v1beta1.ContainerLogLevelOverride {
+	// create a map to store container name and log level data
+	overrideMap := make(map[string]v1beta1.ContainerLogLevelOverride)
+
+	// copy override data to map
+	for _, override := range overrides {
+		if override.LogLevel >= 0 && override.LogLevel <= 10 {
```

Comment:
why check this here? shouldn't this have already been checked in validation func previously?

Reply:
this validation has been removed

Reply:
actually... i dont see any earlier validation... please add some.

Reply:
Validation should happen before setup, in validateRootSync and validateRepoSync

Reply:
validation performed by crd